chore(pipelines/auth): Configure pip to retrieve guarded libraries

EyeCantCU opened the pull request:
Introduce pipeline that configures pip to fetch guarded libraries

EyeCantCU committed:
chore(pipelines/auth): Configure pip to retrieve guarded libraries

Introduces a pipeline that configures the index url for pip with auth
for guarded libraries

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
feat: Add new uv-backed pipelines for Python

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

dannf reviewed (approved):
fwiw, LGTM

dannf reviewed (changes requested):
actually, +1 on the pipeline, but let's not use it for any packages yet. that's an active discussion. But it's useful for testing in the meantime... so, maybe we should require users to set an environment variable to enable it? e.g. `PIP_TESTING_DONT_MERGE_THIS=1` ?

EyeCantCU committed:
chore(auth/pip): Add toggle to disable external index fallback

Adds a 'guarded' toggle that enforces use of Chainguard Libraries by
default. Explicit opt-out is required to fallback to non-Chainguard
Libraries

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
chore(auth/uv): Add toggle to disable external index fallback

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
chore(pipelines/py): Remove venv pipeline

Not needed anymore. We'll install directly to system site packages
instead

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
world: Do not download external Python builds with UV

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
fix(py/install): Various fixes to make this actually work

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
chore(py3-setuptools): Migrate to guarded libraries

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
chore(py3-poetry): Use guarded libraries

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
chore(py/install): Support installation of additional dependencies

Some of our packages are bundles that vendor non-direct dependencies.
Support their installation

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
chore(jupyter-base-notebook): Use guarded libraries where we can

Not all dependencies are available via guarded libraries (i.e.
notebook). So use what we can

And minor cleanup. The CVE remediations weren't actually remediating any
CVEs as they were bumping dependencies in the builder itself and not in
the package

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
chore(superset-4.0): Demo guarded libraries auth with pip

This doesn't fully move superset to guarded libraries, more so for
proving auth with pip

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>
